utils/src/main/java/org/apache/any23/extractor/yaml/YAMLValidator.java

```
@@ -0,0 +1,105 @@
+
+    private static final Pattern YAML_PATTERN = Pattern.compile("^%YAML.*", Pattern.CASE_INSENSITIVE);
+
+    /**
+     * Detects if is contains valid YAML content.
+     * <p>
+     * In the first instance it checks if there is "%YAML" head. If not check
+     * using the brute force method by parsing input stream with yaml parser.
+     * </p>
+     * <p>
+     * NB. Only "false" results are trusted. Even if result is "true" you cannot
+     * be sure that InputStream contains YAML intentional context because
+     * comma-separated-values are pars-able by YAML parser as well.
+     * </p>
+     *
+     * @param is {@link InputStream}
+     * @return
+     * @throws IOException
+     */
+    public static boolean isYAML(InputStream is) throws IOException {
+        if (is == null) {
+            return false;
+        }
+
+        if (!is.markSupported()) {
+            is = new BufferedInputStream(is);
+        }
+
+        boolean result = false;
+
+        // mark the reading frame position. MUST BE FIRST
+        is.mark(Integer.MAX_VALUE);
+
+        while (true) {
+            // if is is empty than return false
+            if (is.available() <= 0) {
+                break;
+            }
+
+            Scanner sc = new Scanner(is);
+            String out = sc.findWithinHorizon(YAML_PATTERN, 0);
+
+            if (out != null && !out.isEmpty()) {
+                log.debug("Head: {}", out);
+                result = true;
+                break;
+            }
+            log.debug("Still not found. output is: {}", out);
+            is.reset();
+
+            try {
+                Yaml yml = new Yaml();
+                Iterable<Object> parsedOut = yml.loadAll(is);
+
+                if (Iterables.size(parsedOut) > 0) {
+                    result = true;
+                    break;
+                }
+            } catch (Exception ex) {
+                //do nothing
```

This should return false rather than silently aborting